stdlib/gui/gui.lua

```diff
@@ -45,12 +47,22 @@ function Gui.Event.dispatch(event)
 
     local gui_element = event.element
     if gui_element and gui_element.valid then
+        local gui_element_name = gui_element.name;
+        local gui_element_state = nil;
+        local gui_element_text = nil;
+
+        if event.name == defines.events.on_gui_checked_state_changed then
+            gui_element_state = gui_element.state
+        end
+
+        if event.name == defines.events.on_gui_text_changed then
+            gui_element_text = gui_element.text
+        end
+
         for gui_element_pattern, handler in pairs(Gui.Event._registry[event.name]) do
-            local match_str = string.match(gui_element.name, gui_element_pattern)
```

Comment:
What about if there are more than one matching patterns for an element? The first one to match destroys the element. Do we then abort the loop ignoring the rest entirely? I chose to store the needed values from the element and pass the info onto all handlers. Let the handlers deal with if their element is still valid or not. Seemed like the better option to me.